### PR TITLE
docs: EXPOSED-670 Adjust YouTrack issue visibility and PR guidelines

### PR DESCRIPTION
**Pull request by bog-walk**
#### Description

**Summary of the change**: Update documentation about YouTrack issue tracker & pull request guidelines.

**Detailed description**:
- **Why**: Issues used to not have public visibility and this was addressed in PR #1798  by adding a note to the README explaining why a 404 message would be met when using an Exposed YouTrack link. Instead, the default visibility of issues has been made public  so none of our links should result in a 404 error for users.

**Other changes:**
- README PR section
    - Updated to include details similar to more extensive list in Contribution page on site
- Contributing page
    - Add more relevant links
    - Use `main` instead of `master`
    - Add YT links for existing issues
    - Add how to format PR title & descriptions
    - Include missing mention about breaking changes
- Added missing KDocs that were not included in a previous PR

---

#### Type of Change

Please mark the relevant options with an "X":
- [X] Documentation update

#### Checklist

- [X] The build is green (including the Detekt check)
- [X] Documentation for my change is up to date

---

#### Related Issues
[EXPOSED-670](https://youtrack.jetbrains.com/issue/EXPOSED-670/Test-project-new-issue-visibility)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,13 +213,15 @@ Do you have questions? Feel free to [request an invitation](https://surveys.jetb
 
 ## Pull requests
 
-We actively welcome your pull requests. However, linking your work to an existing issue is preferred.
+We actively welcome your pull requests. However, linking your work to an [existing issue](https://youtrack.jetbrains.com/issues/EXPOSED) is preferred.
```
Comment (bog-walk) [inline README.md:216]: @vnikolova  It seems like this section has been mostly left alone, while the [website Contributing page](http://jetbrains.github.io/Exposed/contributing.html) has been more thoroughly updated.
I've added some details here but it is still not on the same level as the content in `Contributing.md`. Personally, I would remove this section entirely, and rely on the 2 existing links that lead to the guidelines (1 in 'Links' above and another in 'Contributing' below).
I would even consider moving the 'Contributing' section above the 'Examples' section so it's more visible.
Please let me know what you think should be done with this section.

File: documentation-website/Writerside/topics/Contributing.md [hunk 2]
```diff
@@ -43,21 +43,29 @@ docker context list
...
+1. Fork the [Exposed repository](https://github.com/JetBrains/Exposed), because imitation is the sincerest form of flattery.
 2. Clone your fork to your local machine.
 3. Create a new branch for your changes.
-4. [Create](https://github.com/JetBrains/Exposed/compare) a new PR with a request to merge to the **master** branch.
-5. Ensure that the description is clear and refers to an existing ticket/bug if applicable, prefixing the description with
-   EXPOSED-{NUM}, where {NUM} refers to the YouTrack issue.
+4. [Create](https://github.com/JetBrains/Exposed/compare) a new PR with a request to merge to the **main** branch.
+5. Ensure that the PR title is clear and refers to an [existing ticket/bug](https://youtrack.jetbrains.com/issues/EXPOSED) if applicable,
+   prefixing the title with both a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary)
+   and EXPOSED-\{NUM\}, where \{NUM\} refers to the YouTrack issue code.
```
Comment (bog-walk) [inline documentation-website/Writerside/topics/Contributing.md:54]: In the Writerside preview and on the [website](http://jetbrains.github.io/Exposed/contributing.html#pull-requests), `{NUM}` as it was before will not appear.
I thought escaping `\{NUM\}` would help but it renders without change as `\{NUM\}`.
@vnikolova Do you know how to get just `{NUM}` in markdown?
I could just switch to `[NUM]`, which seems to render properly without needing any escape. Or maybe something like `EXPOSED-###` would be more descriptive?

**Review by vnikolova — APPROVED** [review]
Looks good! I added some minor comments that can be addressed in a separate PR as they are not directly related to the issue this targets.

File: README.md (hunk 1 above)
Comment (bog-walk) [reply README.md:216]: ✅

File: documentation-website/Writerside/topics/Contributing.md (hunk 2 above)
Comment (bog-walk) [reply documentation-website/Writerside/topics/Contributing.md:54]: HTML entity works 🙌

**Pull request merged**